target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-sxr80.dts

```diff
@@ -0,0 +1,643 @@
+		};
+
+		led_power_green: power-green {
+			label = "green:power";
+			gpios = <&tlmm 21 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		led_power_red: power-red {
+			label = "red:power";
+			gpios = <&tlmm 22 GPIO_ACTIVE_LOW>;
+			panic-indicator;
+		};
+
+	};
+};
+
+&tlmm {
+	mdio_pins: mdio-pins {
+		mdc {
+			pins = "gpio68";
+			function = "mdc";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+
+		mdio {
+			pins = "gpio69";
+			function = "mdio";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+	};
+
+	i2c_0_pins: i2c-0-pinmux {
+		mux {
+			pins = "gpio42", "gpio43";
+			function = "blsp1_i2c";
+			drive-strength = <8>;
+			bias-disable;
+		};
+	};
+
+	leds_pins: leds_pinmux {
+		led_power_green {
+			pins = "gpio21";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_power_red {
+			pins = "gpio22";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_white {
+			pins = "gpio26";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_green {
+			pins = "gpio29";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_red {
+			pins = "gpio31";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_blue {
+			pins = "gpio33";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+	};
+
+
+};
+
+&blsp1_uart5 {
+	status = "okay";
+};
+
+&blsp1_i2c2 {
+	pinctrl-0 = <&i2c_0_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+	
+	tlc59208f@27 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "ti,tlc59108";
+		reg = <0x27>;
+		
+		led0@0 {
+			label = "orbi:led0:rgb";
+			reg = <0x0>;
+			linux,default_trigger = "default-on";
+		};
+		led1@1 {
+			label = "orbi:led1:rgb";
+			reg = <0x1>;
+			linux,default_trigger = "default-on";
+		};
+		led2@2 {
+			label = "orbi:led2:rgb";
+			reg = <0x2>;
+			linux,default_trigger = "default-on";
+		};
+		led3@3 {
+			label = "orbi:led3:rgb";
+			reg = <0x3>;
+			linux,default_trigger = "default-on";
+
+		};
+	};
+};
+
+&prng {
+	status = "okay";
+};
+
+&cryptobam {
+	status = "okay";
+};
+
+&crypto {
+	status = "okay";
+};
+
+&qpic_bam {
+	status = "okay";
+};
+
+&qpic_nand {
+	status = "okay";
+
+	nand@0 {
+		reg = <0>;
+		nand-ecc-strength = <4>;
+		nand-ecc-step-size = <512>;
+		nand-bus-width = <8>;
+
+		partitions {
+			compatible = "qcom,smem-part";
+		};
+	};
+};
+
+&mdio {
+	status = "okay";
+
+	pinctrl-0 = <&mdio_pins>;
+	pinctrl-names = "default";
+
+	reset-gpios = <&tlmm 25 1>;
+
+	ethernet-phy@0 {
+		reg = <0>;
+// Disabled to resolve multiple configured reset lines causing issues
+//		reset-gpios = <&tlmm 37 0>;
+	};
+
+	ethernet-phy@1 {
+		reg = <1>;
+	};
+
+	ethernet-phy@2 {
+		reg = <2>;
+	};
+
+	ethernet-phy@3 {
+		reg = <3>;
+	};
+
+	ethernet-phy@4 {
+		reg = <28>;
+// Disabled to resolve multiple configured reset lines causing issues
+//		reset-gpios = <&tlmm 25 1>;
```

Comment:
Same as above, use defines.
Also, I assume this is QCA8081?
If so then utilize the reset-gpios property, try adding a delay after reset and if that doesn't work do it like AX9000 does and set the PHY ID via compatible

Reply:
Alright, I have just wired up my test device again so I can do the changes right now. Also, as this is basically the same as the sxs80 (just a different model number if I am not mistaken) should I use a dtsi for both? How should that dtsi be named? And as the RBR850/RBS850 are also similar-ish (different GPIOs and some different hardware), should they also use the same dtsi then or should I make one for each "pair" of devices?

Reply:
SXS should just be the satellite device?
What's the HW difference between them?

How different are RBX850?
DTSI is great if they are based on the same design and the differences are smaller than just adding complete DTS

Reply:
The SXR80 and SXS80 are the same (it's the satellite, correct), I believe all I need there is a different model number and that's probably it. The only difference is the label of the 2.5Gbit/s ethernet port, on the satellite they are all labeled LAN, on the router that's the WAN port (so I need different IDs to tell them apart in the network.sh stuff).

The RBR850 is kinda similar to the SRX80/SXS80 but has different GPIOs and an additional LED controller. The RBS850 is missing the 2.5Gbit/s port completely, other than that it's a RBR850 (if you buy a phy and ethernet jack and solder them on it would be a RBR850). I'm not really sure about the flash, apparently the SX*80 are using a nand flash, I would have to check what the RB*850 are using.

Reply:
Where did you get the compatible-value from? I just looked up what I wrote in the forum back then: https://forum.openwrt.org/t/adding-openwrt-support-for-xiaomi-ax3600/55049/3041?u=flole

> Yeah, reset is not being released. I know that as the Link is not coming up when I plug a cable in. Once I do it over the sysfs gpio export the link comes up (writing a 1 into it). But I have clearly specified it:

you wrote back then

> I have the exact same issue, its due to upstream kernel not really knowing how to handle multiple GPIO resets.
> Its supposed to support them by PHY as well, but it sometimes works, sometimes not.

As I had to manually de-assert reset when it wasn't declared at the mdio level I don't think it's a timing issue? The link wouldn't even come up. If you believe that it has been fixed I can try removing it from the mdio again and see if that works again. The phy0 is de-asserted by u-boot, so if that reset stuff is broken it would still come online.

Reply:
You should then just make an SXR850.dtsi or similar name and include everything except for the compatible, and have DTS for the router and satellite and just include the DTSI and add compatible and model name.
Then you can use the compatible in the scripts.

As far as RBX850 goes, I don't think they are so similar then and deserve their own DTS.

> Where did you get the compatible-value from?
You make it per the binding:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/Documentation/devicetree/bindings/net/ethernet-phy.yaml?h=v5.10.78#n30

In my case, after I added the compatible it would find the PHY just fine as it was actually deasserting the reset.

Reply:
Fixed